site/source/tutorials/TypeScript_and_Dojo_2.md

```diff
@@ -0,0 +1,60 @@
+
+### Interfaces
+
+Many enterprise-level applications are too large to be developed by a single developer or even a single team. To manage these projects, enterprises seek to divide large projects among multiple teams. Doing this, however, creates challenges as each team must often be able to share [APIs](https://en.wikipedia.org/wiki/Application_programming_interface) and data with other teams. TypeScript's type system supports the concept of an [interface](https://www.typescriptlang.org/docs/handbook/interfaces.html) that can simplify these coordination efforts. In TypeScript, an interface can be used to describe the methods that a class must support or the shape of a data structure. By setting up interfaces between development teams, each team can be confident that their code will be compatible the code from other teams.
+
+In addition to supporting team-coordination, the ability to describe a data-structure's expected *shape* clarifies the expected fields within [object literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#Object_literals). This is especially important when creating applications that interact with a server to load, manipulate, and save large amounts of data. Since most communication between Dojo 2 applications and servers is done using the [JSON protocol](http://www.json.org/), the ability to define interfaces for the messages flowing back and forth greatly simplifies the processing of that data.
+
+### The compiler
+
+Due to the complicated suite of applications within many enterprises, technology upgrades are often adopted slowly. This is necessary due to the large amount of testing that is required to ensure that the new technology does not break existing applications. This has historically hindered enterprise developers from accessing the latest features for the languages that they work in. TypeScript takes advantage of the fact that it must be compiled into JavaScript to bridge this technology gap. TypeScript is continually updated to include the latest features that are coming to the JavaScript language. The compiler then emits JavaScript that converts those features into language structures that are compatible with existing versions of the JavaScript specification, often back to ECMAScript 3. This brings the best of both worlds to the enterprise - they are able to maintain a carefully planned upgrade path for their web browsers and other technology while allowing their developers to take advantage of the latest features of the language.
+
+## Summary
+
+TypeScript's type system has allowed powerful development aids to be created such as refactoring tools and autocompletion support. This enables enterprise developers to focus on writing clean, efficient code without constantly having to lookup information on how to use existing APIs.
```

Comment:
"efficient code without constantly having to lookup information on how to use existing APIs"  -> "efficient code without having to memorize or look up the details for existing APIs"